src/frontend/src/CustomNodes/GenericNode/components/NodeOutputfield/index.tsx

```diff
@@ -0,0 +1,254 @@
+
+  if (outputProxy) {
+    console.log(logHasMessage(flowPoolNode?.data, internalOutputName));
+  }
+
+  const preventDefault = true;
+
+  function handleOutputWShortcut() {
+    if (!displayOutputPreview || unknownOutput) return;
+    if (isThereModal() && !openOutputModal) return;
+    if (selected) {
+      setOpenOutputModal((state) => !state);
+    }
+  }
+
+  const output = useShortcutsStore((state) => state.output);
+  useHotkeys(output, handleOutputWShortcut, { preventDefault });
+
+  let disabledOutput =
+    edges.some((edge) => edge.sourceHandle === scapedJSONStringfy(id)) ?? false;
+
+  function renderTitle() {
+    return (
+      <OutputComponent
+        proxy={outputProxy}
+        idx={index}
+        types={type?.split("|") ?? []}
+        selected={
+          data.node?.outputs![index].selected ??
+          data.node?.outputs![index].types[0] ??
+          title
+        }
+        nodeId={data.id}
+        frozen={data.node?.frozen}
+        name={title ?? type}
+      />
+    );
+  }
+  const handleUpdateOutputHide = (value?: boolean) => {
+    setNode(data.id, (oldNode) => {
+      let newNode = cloneDeep(oldNode);
+      newNode.data = {
+        ...newNode.data,
+        node: {
+          ...newNode.data.node,
+          outputs: newNode.data.node.outputs?.map((output, i) => {
+            if (i === index) {
+              output.hidden = value ?? !output.hidden;
+            }
+            return output;
+          }),
+        },
+      };
+      return newNode;
+    });
+    updateNodeInternals(data.id);
+  };
+
+  useEffect(() => {
+    if (disabledOutput && data.node?.outputs![index].hidden) {
+      handleUpdateOutputHide(false);
+    }
+  }, [disabledOutput]);
+
+  return !showNode ? (
+    <HandleRenderComponent
+      left={false}
+      nodes={nodes}
+      tooltipTitle={tooltipTitle}
+      id={id}
+      title={title}
+      edges={edges}
+      myData={myData}
+      colors={colors}
+      setFilterEdge={setFilterEdge}
+      showNode={showNode}
+      testIdComplement={`${data?.type?.toLowerCase()}-noshownode`}
+    />
+  ) : (
+    <div
+      ref={ref}
+      className="relative mt-1 flex w-full flex-wrap items-center justify-between bg-muted px-5 py-2"
+    >
+      <>
+        <div className="flex w-full items-center justify-end truncate text-sm">
+          <div className="flex-1">
+            <Button
+              disabled={disabledOutput}
+              unstyled
+              onClick={() => handleUpdateOutputHide()}
+              data-testid={`input-inspection-${title.toLowerCase()}`}
+            >
+              <IconComponent
+                className={cn(
+                  "h-4 w-4",
+                  disabledOutput ? "text-muted-foreground" : "",
+                )}
+                strokeWidth={1.5}
+                name={data.node?.outputs![index].hidden ? "EyeOff" : "Eye"}
+              />
+            </Button>
+          </div>
+          <Case condition={data.node?.frozen}>
```

Comment:
what do you think about removing the Case component and just putting the condition?

Reply:
totally agree